src/LegolasFlux.jl

```diff
@@ -79,12 +78,19 @@ ArrowTypes.arrowname(::Type{<:Weights}) = WEIGHTS_ARROW_NAME
 ArrowTypes.JuliaType(::Val{WEIGHTS_ARROW_NAME}) = Weights
 
 #####
-##### `ModelRow`
+##### `ModelV1`
 #####
 
-const ModelRow = Legolas.@row("legolas-flux.model@1",
-                              weights::Union{Missing,Weights} = ismissing(weights) ? missing : Weights(weights),
-                              architecture_version::Union{Missing,Int})
+@schema "legolas-flux.model" Model
+
+@version ModelV1 begin
+    weights::(<:Union{Missing,Weights}) = ismissing(weights) ? missing : Weights(weights)
```

Comment:
this is to parametrize the structs type on the weights, which I think is probably worthwhile (and Legolas pre-v0.5 did)

Reply:
let's leave a comment to that effect with a link the legolas docs?



```suggestion
    # expose the concrete type of `weights` as a type parameter of `ModelV1`
    # see https://github.com/beacon-biosignals/Legolas.jl/blob/main/examples/tour.jl#L291
    weights::(<:Union{Missing,Weights}) = ismissing(weights) ? missing : Weights(weights)
```